app/jobs/assignment_repo/create_github_repository_job.rb

```diff
@@ -67,15 +67,20 @@ def perform(assignment, user)
         )
       end
     rescue Creator::Result::Error => err
-      creator.delete_github_repository(assignment_repo.try(:github_repo_id))
+      creator.delete_github_repository(assignment_repo&.github_repo_id)
       invite_status.errored_creating_repo!
       ActionCable.server.broadcast(
         RepositoryCreationStatusChannel.channel(user_id: user.id),
         text: err,
         status: invite_status.status
       )
       logger.warn(err.message)
-      GitHubClassroom.statsd.increment("v2_exercise_repo.create.fail")
+      case err.message
+      when Creator::REPOSITORY_COLLABORATOR_ADDITION_FAILED
+        GitHubClassroom.statsd.increment("v2_exercise_repo.create.adding_collaborator.fail")
+      else
+        GitHubClassroom.statsd.increment("v2_exercise_repo.create.fail")
+      end
```

Comment:
Are there other cases that we should consider handling differently? Like when starting to import starter code fails?

Reply:
Generally, as much metric granularity as possible is good, because we can aggregate in datadog. I think this is fine for this PR.